components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/util/JdbcUtils.java

```diff
@@ -18,130 +18,143 @@
 
 import org.wso2.carbon.database.utils.jdbc.JdbcTemplate;
 import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
-import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
-
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.DB2;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.H2;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.INFORMIX;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.MARIA_DB;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.MICROSOFT;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.MY_SQL;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.POSTGRE_SQL;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.S_MICROSOFT;
-import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ORACLE;
 
 /**
  * A util class to support the Jdbc executions.
+ * @deprecated use {@link org.wso2.carbon.identity.core.util.JdbcUtils} instead.
  */
 public class JdbcUtils {
 
     /**
      * Get a new Jdbc Template.
      *
      * @return a new Jdbc Template.
+     * @deprecated use {@link org.wso2.carbon.identity.core.util.JdbcUtils#getNewTemplate()} instead.
      */
+    @Deprecated
     public static JdbcTemplate getNewTemplate() {
 
-        return new JdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        return org.wso2.carbon.identity.core.util.JdbcUtils.getNewTemplate();
     }
 
     /**
      * Check if the DB is H2, MySQL or Postgres.
      *
      * @return true if DB is H2, MySQL or Postgres, false otherwise.
      * @throws DataAccessException if error occurred while checking the DB metadata.
+     * @deprecated
      */
+    @Deprecated
     public static boolean isH2MySqlOrPostgresDB() throws DataAccessException {
 
-        return isDBTypeOf(MY_SQL) || isDBTypeOf(H2) || isDBTypeOf(POSTGRE_SQL) || isDBTypeOf(MARIA_DB);
+        return org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB() ||
+                org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB();
```

Comment:
shall we add static imports for these?

Reply:
Addressed by https://github.com/wso2/carbon-identity-framework/pull/3993.